includes/polling/applications/nvidia.inc.php

```diff
@@ -60,6 +61,7 @@
 
         $tags = array('name' => $name, 'app_id' => $app_id, 'rrd_def' => $rrd_def, 'rrd_name' => $rrd_name);
         data_update($device, 'app', $tags, $fields);
-
-    $int++;
 }
+$sm_average = ($sm_total ? 0 : ($sm_total / count($gpuArray)));
```

Comment:
This should probably be the other way around, if `$sm_total` evaluates to true, it should calculate the average.